db/version_set.cc

```diff
@@ -4044,6 +4039,14 @@ Status VersionSet::ProcessManifestWrites(
     mu->Lock();
   }
 
+  if (!io_s.ok()) {
```

Comment:
So the general fix logic is, io_status_ is only updated here. In other places, we use local variable to keep the status in e.g., io_s.

Reply:
Yes, access to io_status_ should be protected by the db_mutex.

Reply:
Yeah, I notice that WriteCurrentStateToManifest is not protected by the db_mutex